packages/adapter-node/src/index.js

```diff
@@ -77,19 +84,19 @@ server.server.on(
 		req.on('close', () => {
 			requests--;
 
-			if (requests === 0 && shutdown_timeout_id) {
-				// when all requests are done, close the connections, so the app shuts down without delay
+			if (shutdown_timeout_id) {
+				// close connections as soon as they become idle, so the app shuts down without delay
```

Comment:
I don't understand exactly what this is doing. If we call `.close()` isn't that closing idle connections already? Why handle it here as well?

Reply:
Apparently `close()` does not close connections that are idle but were opened with a keep alive header until they time out. That's one of the reasons why `closeIdleConnections` was added to Node.js. So to make sure that all idle connections are closed as soon as possible we should call both methods, see also https://github.com/nodejs/node/issues/41578

Reply:
Btw. I added that commit today because it occurred to me that if we don't close the connections asap it could theoretically happen that a connection stays open, continues to receive requests and therefor later has to be forcefully closed.